src/contracts/yas_pool.cairo

```diff
@@ -104,5 +120,91 @@ mod YASPool {
+        {
+            check_ticks(params.tick_lower, params.tick_upper);
+
+            let slot_0 = self.slot_0.read();
+
+            let position = update_position(
+                params.owner,
+                params.tick_lower,
+                params.tick_upper,
+                params.liquidity_delta,
+                slot_0.tick
+            );
+
+            let mut amount_0 = Zeroable::zero();
+            let mut amount_1 = Zeroable::zero();
+            if params.liquidity_delta.is_non_zero() {
+                if slot_0.tick < params.tick_lower {
+                    // current tick is below the passed range; liquidity can only become in range by crossing from left to
+                    // right, when we'll need _more_ token0 (it's becoming more valuable) so user must provide it
+                    amount_0 =
+                        SqrtPriceMath::get_amount_0_delta_signed_token(
+                            TickMath::get_sqrt_ratio_at_tick(params.tick_lower),
+                            TickMath::get_sqrt_ratio_at_tick(params.tick_upper),
+                            params.liquidity_delta
+                        );
+                } else if slot_0.tick < params.tick_upper {
+                    // current tick is inside the passed range
+
+                    amount_0 =
+                        SqrtPriceMath::get_amount_0_delta_signed_token(
+                            slot_0.sqrt_price_X96,
+                            TickMath::get_sqrt_ratio_at_tick(params.tick_upper),
+                            params.liquidity_delta
+                        );
+                    amount_1 =
+                        SqrtPriceMath::get_amount_1_delta_signed_token(
+                            TickMath::get_sqrt_ratio_at_tick(params.tick_lower),
+                            slot_0.sqrt_price_X96,
+                            params.liquidity_delta
+                        );
+
+                    let mut liquidity = self.liquidity.read();
+                    liquidity = LiquidityMath::add_delta(liquidity, params.liquidity_delta);
+                    self.liquidity.write(liquidity);
+                } else {
+                    // current tick is above the passed range; liquidity can only become in range by crossing from right to
+                    // left, when we'll need _more_ token1 (it's becoming more valuable) so user must provide it
+                    amount_1 =
+                        SqrtPriceMath::get_amount_1_delta_signed_token(
+                            TickMath::get_sqrt_ratio_at_tick(params.tick_lower),
+                            TickMath::get_sqrt_ratio_at_tick(params.tick_upper),
+                            params.liquidity_delta
+                        );
+                }
+            }
+            (position, amount_0, amount_1)
+        }
+    }
+
+    /// @dev Common checks for valid tick inputs.
+    fn check_ticks(tick_lower: i32, tick_upper: i32) {
```

Comment:
What if instead of `panicking` here we return a Result or Error accordingly and do the proper assert in `modify_position`? This would bring more transparency to the behavior of the function and still panicking as soon as possible.

Reply:
✅